.github/workflows/build_docs.yml

```diff
@@ -0,0 +1,76 @@
+name: Build and Deploy Documentation
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'docs/**'
+      - '.github/workflows/build_docs.yml'
+  pull_request:
+    paths:
+      - 'docs/**'
+      - '.github/workflows/build_docs.yml'
+  workflow_dispatch:
```

Comment:
I think this would require calling this workflow from Animate/Goalie/Movement/UM2N, similarly to how we have the "reusable" test workflow here in `docs` that is called by them. I think this can become a pain to maintain so I think it's easier to just trigger this workflow manually when we want to update? Since it's just a press of a button :)

Reply:
Although... now that I think about it. Maybe it would be useful to have an option to call it from branches other than `main`, so that we can easily check in PRs that the formatting etc. in docs looks good before merging them. 

But I guess this would then be a smaller workflow that would only run the demos from that particular repo and not from all 4. What do you think?

Reply:
> I think this would require calling this workflow from Animate/Goalie/Movement/UM2N, similarly to how we have the "reusable" test workflow here in `docs` that is called by them. I think this can become a pain to maintain (e.g. how now you had to open 4 separate PRs to fix the `docs` renaming) so I think it's easier to just trigger this workflow manually when we want to update? Since it's just a press of a button :)

I get that, although in general it's nice if we don't have to remember to do things.


> Although... now that I think about it. Maybe it would be useful to have an option to call it from branches other than `main`, so that we can easily check in PRs that the formatting etc. in docs looks good before merging them.
> 
> But I guess this would then be a smaller workflow that would only run the demos from that particular repo and not from all 4. What do you think?

It would be really nice if each repo had a `build_docs` job that could be triggered to build its part of the docs, but I think it's beyond the scope of this PR. It'd be better to just get this merged first. My original question was just in case you were aware of an easy way to set it up.

Reply:
> I get that, although in general it's nice if we don't have to remember to do things.

Sadly I couldn't find a way for the `docs` repo to "listen" for changes in other repos. But we could set up a workflow with a `schedule` trigger that would check, let's say, every 12 hours if there have been changes in these repos. And if so, then the docs could be rebuilt? 

> It would be really nice if each repo had a `build_docs` job that could be triggered to build its part of the docs, but I think it's beyond the scope of this PR. It'd be better to just get this merged first. My original question was just in case you were aware of an easy way to set it up.

Agreed :) I opened https://github.com/mesh-adaptation/docs/issues/78

Reply:
> Sadly I couldn't find a way for the `docs` repo to "listen" for changes in other repos. But we could set up a workflow with a `schedule` trigger that would check, let's say, every 12 hours if there have been changes in these repos. And if so, then the docs could be rebuilt?

Yeah that could work. A nightly check would probably be sufficient.

Reply:
Sounds good. Opened https://github.com/mesh-adaptation/docs/issues/79